examples/torchrec_dlrm/dlrm_handler.py

```diff
@@ -0,0 +1,147 @@
+
+import torch
+from torchrec.datasets.criteo import DEFAULT_CAT_NAMES
+from torchrec.sparse.jagged_tensor import KeyedJaggedTensor
+
+from ts.torch_handler.base_handler import BaseHandler
+
+logger = logging.getLogger(__name__)
+
+
+class TorchRecDLRMHandler(BaseHandler, ABC):
+    """
+    Handler for TorchRec DLRM example
+    """
+
+    def initialize(self, context):
+        """Initialize function loads the model.pt file and initialized the model object.
+           This version creates and initialized the model on cpu fist and transfers to gpu in a second step to prevent GPU OOM.
+        Args:
+            context (context): It is a JSON Object containing information
+            pertaining to the model artifacts parameters.
+        Raises:
+            RuntimeError: Raises the Runtime error when the model.py is missing
+        """
+        properties = context.system_properties
+
+        # Set device to cpu to prevent GPU OOM errors
+        self.device = "cpu"
+        self.manifest = context.manifest
+
+        model_dir = properties.get("model_dir")
+        model_pt_path = None
+        if "serializedFile" in self.manifest["model"]:
+            serialized_file = self.manifest["model"]["serializedFile"]
+            model_pt_path = os.path.join(model_dir, serialized_file)
+
+        # model def file
+        model_file = self.manifest["model"].get("modelFile", "")
+
+        if not model_file:
+            raise RuntimeError("model.py not specified")
+
+        logger.debug("Loading eager model")
+        self.model = self._load_pickled_model(model_dir, model_file, model_pt_path)
+
+        self.map_location = (
+            "cuda"
+            if torch.cuda.is_available() and properties.get("gpu_id") is not None
+            else "cpu"
+        )
+
+        self.device = torch.device(
+            self.map_location + ":" + str(properties.get("gpu_id"))
+            if torch.cuda.is_available() and properties.get("gpu_id") is not None
+            else self.map_location
+        )
+
+        self.model.to(self.device)
+
+        self.model.eval()
+
+        logger.debug("Model file %s loaded successfully", model_pt_path)
+
+        self.initialized = True
+
+    def preprocess(self, data):
+        """
+        The input values for the DLRM model are twofold. There is a dense part and a sparse part.
+        The sparse part consists of a list of ids where each entry can consist of zero, one or multiple ids.
+        Due to the inconsistency in elements, the sparse part is represented by the KeyJaggedTensor class provided by TorchRec.
+
+        Args:
+            data (str): The input data is in the form of a string
+
+        Returns:
+            Tuple of:
+                (Tensor): Dense features
+                (KeyJaggedTensor): Sparse features
+        """
+
+        line = data[0]
+        batch = line.get("data") or line.get("body")
+
+        if not isinstance(batch, dict):
+            batch = json.loads(batch)
```

Comment:
To my understanding, data is expected to be a batch of inference requests. Each request has to be retrieved from batch (eg. [handler batch example](https://github.com/pytorch/serve/blob/master/ts/torch_handler/vision_handler.py#L38))

Reply:
This can be a batch or a single inference request. The example here is of a single request

Reply:
Implemented the changes @lxning